lib/node.js

```diff
@@ -81,36 +75,181 @@ Node.prototype._loadBitcoind = function(config) {
 
 };
 
+/**
+ * This function will find the common ancestor between the current chain and a forked block,
+ * by moving backwards from the forked block until it meets the current chain.
+ * @param {Block} block - The new tip that forks the current chain.
+ * @param {Function} done - A callback function that is called when complete.
+ */
+Node.prototype._syncBitcoindAncestor = function(block, done) {
+
+  var self = this;
+
+  // The current chain of hashes will likely already be available in a cache.
+  self.chain.getHashes(self.chain.tip.hash, function(err, currentHashes) {
+    if (err) {
+      done(err);
+    }
+
+    // Create a hash map for faster lookups
+    var currentHashesMap = {};
+    var length = currentHashes.length;
+    for (var i = 0; i < length; i++) {
+      currentHashesMap[currentHashes[i]] = true;
+    }
```

Comment:
`_syncBitcoindAncestor()` with 500,000 hashes finding a fork six deep into the chain takes less than ~230ms, and should increase linearly the deeper the fork.